cartoframes/io/carto.py

```diff
@@ -92,28 +92,31 @@ def to_carto(dataframe, table_name, credentials=None, if_exists='fail', geom_col
 
     context_manager = ContextManager(credentials)
 
-    cdf = CartoDataFrame(dataframe, copy=True)
+    gdf = GeoDataFrame(dataframe, copy=True)
+
+    if has_geometry(dataframe):
```

Comment:
Similar to the previous comment in Enrichment, I think we could avoid some checks:
```python
    if geom_col in gdf:
        set_geometry(gdf, geom_col, inplace=True)
    elif has_geometry(dataframe):
        gdf.set_geometry(dataframe.geometry.name, inplace=True)

```

Reply:
:+1: